docs/install.md

```diff
@@ -77,21 +77,37 @@ www.example.com {
 ```
 ## Folder Ownership
 
-During installation, Flarum may request that you make certain directories writable. To allow write access to a directory on Linux, execute the following command:
+During installation, Flarum may request that you make certain directories writable.
+Modern operating systems are generally multi-user, meaning that the user you log in as is not the same as the user FLarum is running as.
+The user that Flarum is running as MUST have read + write access to:
+
+- The root install directory, so Flarum can edit `config.php`.
+- The `storage` subdirectory, so Flarum can edit logs and store cached data.
+- The `assets` subdirectory, so that logos and avatars can be uploaded to the filesystem.
+
+Extensions might require other directories, so you might want to recursively grant write access to the entire Flarum root install directory.
+
+There are several commands you'll need to run in order to set up file permissions. Please note that if your install doesn't show warnings after executing just some of these, you don't need to run the rest.
+
+First, you'll need to allow write access to the directory. On Linux:
 
 ```bash
-chmod 775 /path/to/directory
+chmod 775 -R /path/to/directory
 ```
 
-If Flarum requests write access to both the directory and its contents, you need to add the `-R` flag so that the permissions are updated for all the files and folders within the directory:
+If that isn't enough, you may need to check that your files are owned by the correct group and user. By default, in most Linux distributions `www-data` is the group and user that both PHP and the web server operate under. You'll need to look into the specifics of your distro and web server setup to make sure. You can change the folder ownership in most Linux operating systems by running:
 
 ```bash
-chmod 775 -R /path/to/directory
+chown -R www-data:www-data /path/to/directory
```

Comment:
Interesting thing to note here is that, my experience with ownership has been that the web user should be the group owner, and the actual user (cli) should be the owner, but that might only be applicable with the `package-manger` to still be able to run manual composer commands, and in-app composer commands, I can't actually think of other examples not involving that atm.

Reply:
The way I usually get around this is to add the CLI user to the web group. But that's a good thought.